### PR TITLE
Add CachedDictionary as lazy read only dictionary

### PR DESCRIPTION
**Pull request by chamons**
- Use in two trivial cases as example

### CODE REVIEW

**Comment by chamons** [comment]
If people like this approach, I can update other usages of the pattern.

**Comment by chamons** [comment]
Woops, forgot to update usages of EnsureGetNextPageMethods() / GetOperationMethod.

Please hold.

**Comment by chamons** [comment]
Corrected.

File: src/AutoRest.CSharp/Output/Models/DataPlaneRestClient.cs [hunk 1]
```diff
@@ -190,13 +183,13 @@ private static RestClientMethod BuildNextPageMethod(RestClientMethod method, Ope
 
         public RestClientMethod? GetNextOperationMethod(ServiceRequest request)
         {
-            EnsureGetNextPageMethods().TryGetValue(request, out RestClientMethod? value);
+            _nextPageMethods.TryGetValue(request, out RestClientMethod? value);
```
Comment (pakrym) [inline src/AutoRest.CSharp/Output/Models/DataPlaneRestClient.cs:186]: This doesn't look like an equivalent change.

Comment (pakrym) [reply src/AutoRest.CSharp/Output/Models/DataPlaneRestClient.cs:186]: ah, it does.

**Comment by pakrym** [comment]
I like it but wonder if it would create merge churn for management plane folks.

**Comment by chamons** [comment]
@m-nash - Thoughts on this PR?

I could land it without changing the data plane clients to make merge easier, and have to go back post merge updating them all.

I could also just update everything now, and you consume all of the churn now.

**Comment by chamons** [comment]
Re-running failed tests as they seem to work for me locally. Maybe networking issue?

**Review by pakrym — APPROVED** [review]

**Review by m-nash — APPROVED** [review]
As written this shouldn't affect the merge much at all.

**Pull request merged**